### PR TITLE
Add InputFileContentExtractor, ProfileDefinitionReader, RuleMetadataLoader

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Not designed for multi-threads
+ */
+public final class ProfileDefinitionReader {
+
+  private final RuleFinder ruleFinder;
+  private JsonParser jsonParser;
```
Comment (pynicolas) [inline src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java]: `jsonParser` can be local to `loadActiveKeysFromJsonProfile`.

File: src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.sonar.api.rules.RuleFinder;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Not designed for multi-threads
+ */
+public final class ProfileDefinitionReader {
+
+  private final RuleFinder ruleFinder;
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java]: MINOR Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  private JsonParser jsonParser;
+
+  public ProfileDefinitionReader(RuleFinder ruleFinder) {
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java]: MINOR Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java (hunk 1 above)
Comment (vilchik-elena) [inline src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java]: @pynicolas fixed

File: src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * Use to create {@link RulesProfile} based on json file.
+ *
+ * Not designed for multi-threads
+ */
+public final class ProfileDefinitionReader {
+
+  private final RuleFinder ruleFinder;
+
+  public ProfileDefinitionReader(RuleFinder ruleFinder) {
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/plugin/commons/ProfileDefinitionReader.java]: MINOR Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Pull request merged**